cypress-io/cypress
# fix: update RunResult.stats on return of CLI

Closes https://github.com/cypress-io/cypress/issues/23805

### Additional details

The PR guarantee that the interface RunResult of CypressCommandLine is equal to the returned value of application rules when the listener after:spec is called.

### Steps to test
1. Add listener after:spec on root of application to log spec and results:
```
on('after:spec', (spec, results) => {
  console.log(spec)
  console.log(results)
})
```
2. Run any single e2e test by cypress run
3. Verify in log that the object keys of the `stats` object is the same of stats object declared on `cli/types/cypress-npm-api.d.ts (line 226)`


### PR Tasks
[na] Have tests been added/updated?
[na] Has a PR for user-facing changes been opened in [cypress-documentation](https://github.com/cypress-io/cypress-documentation)?
[na] Have API changes been updated in the [type definitions](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
